### PR TITLE
11510 supports setting private_dns_name on resource_aws_vpc_endpoint_…

### LINKED ISSUE #11510: AWS PrivateLink now supports Private DNS names for internal and 3rd party services
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

You can now access AWS PrivateLink based services privately from within your VPC using [Private DNS names](https://docs.aws.amazon.com/vpc/latest/userguide/vpce-interface.html#vpce-private-dns).

### New or Affected Resource(s)

* [`aws_vpc_endpoint`](https://www.terraform.io/docs/providers/aws/r/vpc_endpoint.html)
* [`aws_vpc_endpoint_service`](https://www.terraform.io/docs/providers/aws/r/vpc_endpoint_service.html)

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2020/01/aws-privatelink-supports-private-dns-names-internal-3rd-party-services/).
[User guide](https://docs.aws.amazon.com/vpc/latest/userguide/vpce-interface.html#vpce-private-dns).

Requires[AWS SDK v1.27.1](https://github.com/aws/aws-sdk-go/releases/tag/v1.27.1):
* https://github.com/terraform-providers/terraform-provider-aws/pull/11501

### PR DESCRIPTION
**Pull request by craigedmunds**
Fixes #11510 by passing through the optional private_dns_name through to the aws sdk PrivateDnsName parameter

### CODE REVIEW

**Comment by bondezbond** [comment]
any idea on when this will be released?

File: aws/resource_aws_vpc_endpoint_service.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ func resourceAwsVpcEndpointService() *schema.Resource {
 			},
 			"private_dns_name": {
 				Type:     schema.TypeString,
-				Computed: true,
```
Comment (ewbankkit) [inline aws/resource_aws_vpc_endpoint_service.go:63]: Leave `Computed: true` here as the value is computed if no explicit DNS name is specified.

File: website/docs/d/vpc_endpoint_service.html.markdown [hunk 2]
```diff
@@ -48,6 +48,7 @@ The given filters must match exactly one VPC endpoint service whose data will be
 
 * `service` - (Optional) The common name of an AWS service (e.g. `s3`).
 * `service_name` - (Optional) The service name that can be specified when creating a VPC endpoint.
+* `private_dns_name` - (Optional) The private DNS name for the service.
```
Comment (ewbankkit) [inline website/docs/d/vpc_endpoint_service.html.markdown:51]: Please remove from the _Attributes Reference_ section below.

**Comment by ewbankkit** [comment]
@craigedmunds Thanks for this.
Are you able to rebase and remove the merge conflict?

**Comment by craigedmunds** [comment]
Hi, comments addressed & rebased.

**Comment by ewbankkit** [comment]
@craigedmunds Thanks for the quick response.
Could you please run `make website-lint-fix` and commit the changes, to make Travis happy.

**Comment by craigedmunds** [comment]
That’s done @ewbankkit

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSVpcEndpointService_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 2 -run=TestAccAWSVpcEndpointService_ -timeout 120m
=== RUN   TestAccAWSVpcEndpointService_basic
=== PAUSE TestAccAWSVpcEndpointService_basic
=== RUN   TestAccAWSVpcEndpointService_PrivateDnsName
=== PAUSE TestAccAWSVpcEndpointService_PrivateDnsName
=== RUN   TestAccAWSVpcEndpointService_AllowedPrincipalsAndTags
=== PAUSE TestAccAWSVpcEndpointService_AllowedPrincipalsAndTags
=== RUN   TestAccAWSVpcEndpointService_removed
=== PAUSE TestAccAWSVpcEndpointService_removed
=== CONT  TestAccAWSVpcEndpointService_basic
=== CONT  TestAccAWSVpcEndpointService_removed
--- PASS: TestAccAWSVpcEndpointService_basic (279.31s)
=== CONT  TestAccAWSVpcEndpointService_AllowedPrincipalsAndTags
--- PASS: TestAccAWSVpcEndpointService_removed (321.97s)
=== CONT  TestAccAWSVpcEndpointService_PrivateDnsName
--- PASS: TestAccAWSVpcEndpointService_AllowedPrincipalsAndTags (333.83s)
--- PASS: TestAccAWSVpcEndpointService_PrivateDnsName (296.19s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	618.216s
```

**Comment by craigedmunds** [comment]
Are you going to merge it @ewbankkit ?

**Comment by bayazee** [comment]
@craigedmunds I guess we should also add domain verification name and value support. Maybe as attributes?

**Comment by Mattias-** [comment]
Good job @craigedmunds thanks for helping out with this!

**Comment by kevholditch** [comment]
@craigedmunds I've fixed the merge conflict and submitted a PR to your repo.  Thanks for the great work hoping to get this merged as need it for a project

**Comment by kevholditch** [comment]
@craigedmunds sorry my fix broke `go fmt` that is fixed now with another PR to your repo

**Comment by patcable** [comment]
Did a little digging to see if I could add the attributes for DNS record validation to @craigedmunds's fork; have a question over in #11510 on the "right way" to do this in TF

**Comment by nickelozz** [comment]
We're also needing private DNS name support in the VPC endpoint service resource. @ewbankkit are you able to provide any additional comments to identify what's missing to get this PR merged?

**Comment by bflad** [comment]
Hi @craigedmunds 👋  Thank you for submitting this. We have opted to pull in #16495, which includes the PrivateDnsNameConfiguration attributes that are returned from the EC2 API and is up to date without merge conflicts. This functionality will release with version 3.23.0 of the Terraform AWS Provider, later this week. 👍

**Pull request closed without merging**